app/models/invoice.rb

```diff
@@ -612,4 +615,15 @@ def notify_buyer_about_payment
       InvoiceMessenger.successfully_charged(self).deliver
     end
   end
+
+  private
+
+  # Allowed range for the invoice period is:
+  # - from: creation date of the buyer
+  # - to: current or the following month - allowing to issue invoices for the future month
+  def period_range_valid?
+    return if self[:period]&.between?(buyer.created_at.to_date.beginning_of_month, Time.zone.now + 1.month)
```

Comment:
I don't think it makes an actual difference but throughout the whole file we use `Time.now.utc` and here `Time.zone.now`. I think we should be consistent.

Reply:
Yeah, I agree... But it's tricky, because the class of the `period` field (Month) uses `Time.zone` :shrug:

Reply:
Indeed, the timestamps (including `created_at`) are stored in the DB in UTC, but when they are pulled from the DB in Rails models, the DateTime type is converted to Rails time zone. So, I think this should be fine (using timezone-aware timestamps everywhere).

Reply:
I was thinking about that `self[:period]` might be the buyer's time zone, not UTC. But I might be wrong.